src/inverted_index.c

```diff
@@ -1205,39 +1204,45 @@ int IndexBlock_Repair(IndexBlock *blk, DocTable *dt, IndexFlags flags, IndexRepa
     // this will close the "hole" in the index
     if (!docExists) {
       if (params->RepairCallback) {
-        params->RepairCallback(res, params->arg);
+        params->RepairCallback(res, blk, params->arg);
+      }
+      if (!frags++) {
+        // First invalid doc; copy everything prior to this to the repair
+        // buffer
+        Buffer_Write(&bw, blk->buf.data, bufBegin - blk->buf.data);
```

Comment:
Why do you need this? They are working on the same buffer

Reply:
No, they aren't working on the same buffer anymore. We make a separate buffer for repairs; however, we ensure that we don't actually allocate memory until a missing doc is found

Reply:
The buffer is separate because we would end up in situations where `memcpy` received overlapping arguments.